File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
        prev = None
        while head:
            temp = head.next
            head.next = prev
            prev = head
            head = temp
        count = 1
        dummy = ListNode()
        curr = dummy
        while prev:
            if count != n:
                curr.next = prev
            else:
                curr.next = prev.next
            curr = prev
            prev = prev.next
            count += 1
        head = dummy.next
        prev = None
        while head:
            temp = head.next
            head.next = prev
            prev = head
            head = temp
        return prev
            
            
            
                
            
        

File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* removeNthFromEnd(ListNode* head, int n) {
        ListNode* curr = head;
        int length = 0;
        while(curr){
            curr = curr->next;
            length ++;
        }
        if(length == 1){
            return NULL;
        }
        int position = length - n + 1;
        if(position == 1){
            return head->next;
        }
        int index = 0;
        curr = head;
        while(index < position - 2){
            curr = curr->next;
            index ++;
        }
        curr->next = curr->next->next;
        return head;
    }
};